blssig/aggregation.go

```diff
@@ -46,19 +49,50 @@ func (v Verifier) VerifyAggregate(msg []byte, signature []byte, pubkeys []gpbft.
 	return v.scheme.Verify(aggPubKey, msg, signature)
 }
 
-func (v Verifier) pubkeysToMask(pubkeys []gpbft.PubKey) (*sign.Mask, error) {
+func (v *Verifier) pubkeyToPoint(p gpbft.PubKey) (kyber.Point, error) {
+	if len(p) > 96 {
+		return nil, fmt.Errorf("public key is too large: %d > 96", len(p))
+	}
+
+	v.mu.RLock()
+	point, ok := v.pointCache[string(p)]
+	v.mu.RUnlock()
+	if ok {
+		return point.Clone(), nil
+	}
+
+	point = v.keyGroup.Point()
+	err := point.UnmarshalBinary(p)
+	if err != nil {
+		return nil, fmt.Errorf("unarshalling pubkey: %w", err)
+	}
+	if point.Equal(v.keyGroup.Point().Null()) {
+		return nil, fmt.Errorf("public key is a null point")
+	}
+	v.mu.Lock()
+
+	// Initialize the cache, or re-initialize it if we've grown too big. We don't expect the
+	// latter to happen in practice (would need over 10k participants), but better be safe than
+	// sorry. We could, alternatively, use an LRU but... that's not worth the overhead for
+	// somethign that shouldn't happen.
+	if v.pointCache == nil || len(v.pointCache) >= maxPointCacheSize {
+		v.pointCache = make(map[string]kyber.Point)
```

Comment:
How much churn do we get in participants? 

It doesn't matter much since the cache size is capped but I am curious if it is worth thinking about some eviction strategy to avoid sudden change in performance characteristics should we reach the max size.

Reply:
Very little, that's why I stuck with a simple cache instead of trying to come up with a nice eviction strategy. The main source of churn will be SPs changing keys.

NOTE: if we hit >10k participants, _then_ this cache will become useless.